labgrid/driver/networkusbstoragedriver.py

```diff
@@ -49,6 +51,52 @@ def on_activate(self):
     def on_deactivate(self):
         pass
 
+    @step(args=['filenames', 'target_dir', 'partition'])
+    def copy_files(self, filenames, target_dir, partition=1):
+        if not self.storage.path:
+            raise ExecutionError(
+                "{} is not available".format(self.storage_path)
+            )
+
+        dev_path = pathlib.PurePath(self.storage.path + str(partition))
+        mount_path = pathlib.PurePath('/media') / dev_path.name
+
+        mount_args = ["pmount", dev_path]
+        self.logger.debug('Mount {} to {}'.format(dev_path, mount_path))
+        try:
+            processwrapper.check_output(
+                self.storage.command_prefix + mount_args)
+        except subprocess.CalledProcessError as e:
+            # Proceed if device already has been mounted with pmount, otherwise
+            # give up
+            if not (e.returncode == 4 and str(mount_path) in str(e.output)):
+                raise e
+
+        target_path = pathlib.PurePath(mount_path) / target_dir
+        for f in filenames:
+            mf = ManagedFile(f, self.storage)
+            mf.sync_to_resource()
+            self.logger.debug("Copy {} to {}".format(
+                mf.get_remote_path(), target_path))
+            cp_args = ["cp", mf.get_remote_path(), "-t", target_path]
+            processwrapper.check_output(self.storage.command_prefix + cp_args)
```

Comment:
This whole block should probably go into a try/finally, to ensure that the device is unmounted even in error cases.

Reply:
Fixed